### PR TITLE
7 post comments

### PR DESCRIPTION
**Pull request by natccc**
add /api/articles/:article_id/comments endpoint and the tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -102,46 +102,102 @@ describe("/api/articles", () => {
...
+  });
+  it("GET 400: returns error for bad id", () => {
+    return request(app)
+      .get("/api/articles/dog/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  it("POST 201: responds with the posted comment", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:147]: great clear test descriptions

File: __tests__/app.test.js [hunk 2]
```diff
@@ -102,46 +102,102 @@ describe("/api/articles", () => {
...
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "butter_bridge",
+        body: "This is a test comment"
+      })
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        console.log(comment)
```
Comment (roisinoreilly) [inline __tests__/app.test.js:156]: don't forget to remove any console.logs

File: __tests__/app.test.js [hunk 3]
```diff
@@ -102,46 +102,102 @@ describe("/api/articles", () => {
...
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  it('POST 400: returns error for posting incorrect column names', () => {
+    return request(app)
+     .post("/api/articles/1/comments")
+     .send({
+        banana: "butter_bridge",
```
Comment (roisinoreilly) [inline __tests__/app.test.js:195]: great edge case

**Review by roisinoreilly — COMMENTED** [review]
great work again. I would just like to see an extra test for if you try and post a comment but the username isn't in the database (this should fail), but otherwise really nice job
